src/vdom/component.js

```diff
@@ -144,12 +144,18 @@ export function renderComponent(component, opts, mountAll, isChild) {
 		if (initialBase && base!==initialBase && inst!==initialChildComponent) {
 			let baseParent = initialBase.parentNode;
 			if (baseParent && base!==baseParent) {
-				baseParent.replaceChild(base, initialBase);
 
 				if (!toUnmount) {
+					baseParent.replaceChild(base, initialBase);
 					initialBase._component = null;
 					recollectNodeTree(initialBase, false);
 				}
+				// @TODO Formerly the initialBase was always replaced by the base
```

Comment:
Yeah, this was the issue. My solution was to prepend the new element before the existing one, as this gets removed in the ‘unmountComponent’ function. But it seems, that the replacements is additionally done later in the stack. This would explain the passing tests.